framework/include/userobjects/LayeredSideIntegralBase.h

```diff
@@ -0,0 +1,112 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "LayeredBase.h"
+#include "InputParameters.h"
+
+/**
+ * Base class for computing layered side integrals
+ */
+template <typename UserObjectType>
+class LayeredSideIntegralBase : public UserObjectType, public LayeredBase
+{
+public:
+  static InputParameters validParams();
+
+  LayeredSideIntegralBase(const InputParameters & parameters);
+
+  /**
+   * Given a Point return the integral value associated with the layer that point falls in.
+   *
+   * @param p The point to look for in the layers.
+   */
+  virtual Real spatialValue(const Point & p) const override { return integralValue(p); }
+
+  virtual const std::vector<Point> spatialPoints() const override;
```

Comment:
I would not return as `const` value, but maybe this unfortunate pattern has already propagated into apps?

Reply:
No idea, I just blindly moved everything into a base class...

Reply:
we'll say this isn't this PR's circus